### ISSUE TITLE
All commands hang with 100% CPU [if have only 1 CPU core]

### ISSUE THREAD

**Issue by dalee-bis** [Azure/azure-storage-azcopy#419]
##### Note: The version is visible when running AzCopy without any argument

AzCopy 10.1.2

### Which platform are you using? (ex: Windows, Mac, Linux)

* New Azure VM (nothing installed)
* Windows Server 2016 Datacenter
* Standard DS1 v2 (1 vcpus, 3.5 GiB memory)
* With an identity and permissions to access the storage account (trying to test permissions)

##### Note: Please remove the SAS to avoid exposing your credentials. If you cannot remember the exact command, please retrieve it from the beginning of the log file.

```
azcopy login
azcopy sync 'C:\Users\me\UploadTest\' https://uploadpermtest.blob.core.windows.net/test/ --recursive --log-level INFO
azcopy list https://uploadpermtest.blob.core.windows.net/test/
```

### What problem was encountered?

Other then when commands are incorrect and help text is displayed, all commands immediately hang with no output and use 100% CPU. Adding the log level argument has no effect.

The only output is a `C:\Users\me\.azcopy\plans` folder which is empty.

### How can we reproduce the problem in the simplest way?

Create an Azure VM matching the setup above, download AzCopy, execute `azcopy login`.

### Have you found a mitigation/solution?

No

**Comment by zezha-msft** [comment]
Hi @dalee-bis, thanks for reaching out!

It is a known issue that AzCopy has trouble in low memory environments. We have it on our backlog to look into.

**Comment by JohnRusk** [comment]
@zezha-msft I ran a few tests. It looks very much like the issues is not memory related, but rather is related to core count. Specifically:
* The failure can be observed with AzCopy login, which does not touch the parts of the code that we have already earmarked for memory usage improvement.
* When I reproduced it on the same VM size as @dalee-bis, Task Manager showed very high CPU usage, but very low RAM usage.
* I can also reproduce it on a much larger VM, by using a CPU affinity mask to restrict AzCopy to just one CPU core. Change the mask to allow the process to use two cores, and the problem goes away.

@dalee-bis, due to the above, I'm going to add an suffix about core count to the issue title.
